cmd/flux/cmd/test.go

```diff
@@ -0,0 +1,151 @@
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/ast"
+	"github.com/influxdata/flux/ast/edit"
+	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/execute/executetest"
+	"github.com/influxdata/flux/execute/table"
+	"github.com/influxdata/flux/fluxinit"
+	"github.com/influxdata/flux/internal/errors"
+	"github.com/influxdata/flux/lang"
+	"github.com/influxdata/flux/memory"
+	"github.com/influxdata/flux/parser"
+	"github.com/influxdata/flux/runtime"
+	"github.com/spf13/cobra"
+)
+
+type Test struct {
+	ast *ast.Package
+	Err error
+}
+
+func NewTest(ast *ast.Package) Test {
+	return Test{
+		ast: ast,
+	}
+}
+
+func (t *Test) Name() string {
+	return t.ast.Files[0].Name
+}
+
+func (t *Test) Run() {
+	jsonAST, Err := json.Marshal(t.ast)
+	if Err != nil {
+		t.Err = Err
+		return
+	}
+	c := lang.ASTCompiler{AST: jsonAST}
+
+	ctx := executetest.NewTestExecuteDependencies().Inject(context.Background())
+	program, Err := c.Compile(ctx, runtime.Default)
+	if Err != nil {
+		t.Err = errors.Wrap(Err, codes.Invalid, "failed to compile")
+	}
+
+	alloc := &memory.Allocator{}
+	result, Err := program.Start(ctx, alloc)
+	if Err != nil {
+		// XXX: rockstar (8 Dec 2020) - Not all tests should return streaming data.
+		if !strings.Contains(Err.Error(), "this Flux script returns no streaming data") {
+			t.Err = errors.Wrap(Err, codes.Inherit, "error while executing program")
+			return
+		}
+	}
+	defer result.Done()
+
+	for res := range result.Results() {
+		Err := res.Tables().Do(func(tbl flux.Table) error {
+			// If there *is* streaming data from the flux test, it is assumed to come from
+			// `testing.diff`, and if that returns tables, that they are showing the failed diff.
+			// XXX: rockstar (08 Dec 2020) - This could use some ergonomic work, as the diff output
+			// is not exactly "human readable."
+			return fmt.Errorf("%s", table.Stringify(tbl))
+		})
+		if Err != nil {
+			t.Err = Err
+		}
+	}
+}
+
+var testCommand = &cobra.Command{
+	Use:   "test",
+	Short: "Run flux tests",
+	Long:  "Run flux tests",
+	Run: func(cmd *cobra.Command, args []string) {
+		fluxinit.FluxInit()
+		runFluxTests()
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(testCommand)
+}
+
+func runFluxTests() {
+	root, Err := filepath.Abs("./stdlib")
```

Comment:
Can you change this one part to be an input argument? I'd like to see this be a list of filepaths or files to test if it's possible. So we could do:

```
flux test ./stdlib
```

If you want, we can delay this until #3334 is implemented since that touches on that issue too.

Reply:
Yeah, I think this should be a command line argument, but I wanted to leave that to a followup patch. I do agree with you though.

Reply:
Then we'll wait for the follow-up PR.